lib/udf_findhotel_utils.rb

```diff
@@ -7,25 +7,33 @@ class UdfFindhotelUtils
           params:      "component varchar(max), room varchar(max)",
           return_type: "integer",
           body:        %~
+
+            import logging
+            logger = logging.getLogger('parse_room_description')
+
             if not room:
               return None
             else:
-              adults = 0
-              children = 0
-              rooms = room.split('|')
-              for r in rooms:
-                ch = r.split(':')
-                adults += int(ch[0])
-                if len(ch) > 1:
-                  children += len(ch[-1].split(','))
+              try:
+                adults = 0
+                children = 0
+                rooms = room.split('|')
+                for r in rooms:
+                  ch = r.split(':')
+                  adults += int(ch[0])
+                  if len(ch) > 1:
+                    children += len(ch[-1].split(','))
 
-              if component.lower() == 'adults':
-                return adults
-              elif component.lower() == 'children':
-                return children
-              elif component.lower() == 'rooms':
-                return len(rooms)
-              else:
+                if component.lower() == 'adults':
+                  return adults
+                elif component.lower() == 'children':
+                  return children
+                elif component.lower() == 'rooms':
+                  return len(rooms)
+                else:
+                  return None
+              except (ValueError, IndexError) as e:
+                logger.error("Error: " + str(e) + " - Invalid room query " + str(room))
```

Comment:
You're welcome.

Weird for the logging because both `%` and `{` should work. Can you give a last try to that syntax pls:
`logger.error("Error %s - Invalid room query %s", e, room)`
And if it still doesn't work never mind, you can merge it like that it is no big thing.